src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -628,6 +630,50 @@ public override string GetName(int ordinal)
 
         #endregion
 
+        #region Generic value getters
+
+        internal delegate T ReadDelegate<T>(NpgsqlReadBuffer buffer, int columnLen, FieldDescription fieldDescription);
+
+        internal delegate ValueTask<T> ReadAsyncDelegate<T>(NpgsqlReadBuffer buffer, int columnLen, bool async, FieldDescription fieldDescription);
+
+        internal static class NullableHandler<T>
```

Comment:
This approach is great, especially compared to the previous ConcurrentDictionary approach.